smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/GradleGenerator.kt

```diff
@@ -9,78 +9,202 @@ import software.amazon.smithy.kotlin.codegen.KotlinSettings
-        write("implementation(kotlin(\"stdlib\"))")
-        // TODO - can we make kotlin dependencies not specify a version e.g. kotlin("kotlin-test")
-        // TODO - Kotlin MPP setup (pass through KotlinSettings) - maybe separate gradle writers
-        val orderedDependencies = dependencies.sortedWith(compareBy({ it.config }, { it.artifact }))
-        for (dependency in orderedDependencies) {
-            write("${dependency.config}(\"#L:#L:#L\")", dependency.group, dependency.artifact, dependency.version)
+            write(formatted)
         }
     }
 
-    val annotations = settings.build.optInAnnotations
-    if (annotations != null && annotations.isNotEmpty()) {
-        writer.openBlock("val optinAnnotations = listOf(")
-            .call {
-                val formatted = annotations.joinToString(
-                    separator = ",\n",
-                    transform = {
-                        "\"$it\""
-                    }
-                )
+    val pluginsBodyRenderer: InlineCodeWriter = {
+        val pluginName = if (isKmp) "multiplatform" else "jvm"
 
-                writer.write(formatted)
+        write(
+            "kotlin(\"$pluginName\") #W",
+            { w: CodeWriter ->
+                if (isRootModule) {
+                    w.write("""version "#L"""", KOTLIN_COMPILER_VERSION)
+                }
             }
-            .closeBlock(")")
-
-        writer.openBlock("kotlin.sourceSets.all {")
-            .write("optinAnnotations.forEach { languageSettings.optIn(it) } ")
-            .closeBlock("}")
+        )
     }
 
-    writer.write("")
-        .openBlock("tasks.test {")
-        .write("useJUnitPlatform()")
-        .openBlock("testLogging {")
-        .write("""events("passed", "skipped", "failed")""")
-        .write("showStandardStreams = true")
-        .closeBlock("}")
-        .closeBlock("}")
+    when {
+        isKmp -> renderKmpGradleBuild(
+            writer,
+            isRootModule,
+            dependencies,
+            pluginsBodyRenderer,
+            repositoryRenderer,
+            annotationRenderer
+        )
+        else -> renderJvmGradleBuild(
+            writer,
+            isRootModule,
+            dependencies,
+            pluginsBodyRenderer,
+            repositoryRenderer,
+            annotationRenderer
+        )
+    }
 
     val contents = writer.toString()
     manifest.writeFile("build.gradle.kts", contents)
     if (settings.build.generateFullProject) {
         manifest.writeFile("settings.gradle.kts", "")
     }
 }
+
+fun renderKmpGradleBuild(
+    writer: CodeWriter,
+    isRootModule: Boolean,
+    dependencies: List<KotlinDependency>,
+    pluginsRenderer: InlineCodeWriter,
+    repositoryRenderer: InlineCodeWriter,
+    annotationRenderer: InlineCodeWriter
+) {
+    writer.write(
+        """
+        plugins {
+            #W
+        }  
+        #W
+        
+        kotlin {
+            jvm {
+                compilations.all {
```

Comment:
question

Should this entire block be dependent on whether we generate the root module or not?

In general when not generating the root module we have been sort of just relying on the parent project to take care of specific configuration like this. 

e.g. for non root module we maybe just generate:

```kotlin
plugins {... }

kotlin {
    jvm()
    
    sourceSets { ... }

    ...
}

```

I'm not sure it matters right now but it would align with what we have been doing.